cpp/src/jni/dataset/jni_util.cc

```diff
@@ -162,6 +165,15 @@ std::shared_ptr<ReservationListener> ReservationListenableMemoryPool::get_listen
 
 ReservationListenableMemoryPool::~ReservationListenableMemoryPool() {}
 
+Status CheckException(JNIEnv* env) {
+  if (env->ExceptionCheck()) {
+    env->ExceptionDescribe();
+    env->ExceptionClear();
+    return Status::Invalid("Error during calling Java code from native code");
```

Comment:
Instead of simply dumping the exception, is it possible to add its description to the `Status` instance that is being returned?

Reply:
This can be deferred to a later JIRA btw.

Reply:
Sure. I'll open another ticket for that.

Reply:
Ok, I opened https://issues.apache.org/jira/browse/ARROW-16329

Reply:
Assigned to me. Thanks.